portfolio/src/main/java/com/google/sps/servlets/NewCommentServlet.java

```diff
@@ -38,10 +40,18 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     Entity commentEntity = new Entity("Comment");
     commentEntity.setProperty("timestamp", timestamp);
     commentEntity.setProperty("comment", comment);
+    commentEntity.setProperty("id", getId());
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     datastore.put(commentEntity);
 
     response.sendRedirect("/");
   }
+
+  /** Returns the user's id */
+  private String getId(){
+    UserService userService = UserServiceFactory.getUserService();
+    String id = userService.getCurrentUser().getUserId();
```

Comment:
nit: return without creating a new variable.
```suggestion
    return userService.getCurrentUser().getUserId();
```